times_reader/main.py

```diff
@@ -329,6 +329,173 @@ def produce_times_tables(
     return result
 
 
+# TODO should this be a part of the restructured mapping file?
+table_order = [
+    "ALL_TS",
+    "ALL_REG",
+    "REG",
+    "COM_GRP",
+    "COM",
+    "PRC",
+    "CUR",
+    "UNITS",
+    "UNITS_COM",
+    "UNITS_CAP",
+    "UNITS_ACT",
+    "UNITS_MONY",
+    "G_DYEAR",
+    "COM_DESC",
+    "COM_GMAP",
+    "COM_LIM",
+    "COM_TMAP",
+    "COM_TSL",
+    "COM_UNIT",
+    "NRG_TMAP",
+    "PRC_ACTUNT",
+    "PRC_DESC",
+    "PRC_MAP",
+    "PRC_NOFF",
+    "PRC_TSL",
+    "PRC_VINT",
+    "PRC_DSCNCAP",
+    "TS_GROUP",
+    "TS_MAP",
+    "DATAYEAR",
+    "PASTYEAR",
+    "MODLYEAR",
+    "TOP",
+    "TOP_IRE",
+    "COM_PEAK",
+    "COM_PKTS",
+    "UC_N",
+    "UC_R_SUM",
+    "UC_R_EACH",
+    "UC_ATTR",
+    "B",
+    "E",
+    "ACT_BND",
+    "ACT_COST",
+    "ACT_CUM",
+    "NCAP_AF",
+    "NCAP_AFA",
+    "NCAP_BND",
+    "NCAP_CHPR",
+    "NCAP_COST",
+    "NCAP_CPX",
+    "NCAP_DRATE",
+    "NCAP_ELIFE",
+    "NCAP_FOM",
+    "NCAP_ILED",
+    "NCAP_PASTI",
+    "NCAP_TLIFE",
+    "NCAP_START",
+    "CAP_BND",
+    "COM_BNDNET",
+    "COM_CSTNET",
+    "COM_FR",
+    "COM_IE",
+    "COM_TAXNET",
+    "COM_AGG",
+    "COM_ELAST",
+    "COM_PROJ",
+    "COM_STEP",
+    "COM_VOC",
+    "FLO_COST",
+    "FLO_DELIV",
+    "FLO_FR",
+    "FLO_FUNC",
+    "FLO_FUNCX",
+    "FLO_SHAR",
+    "FLO_SUB",
+    "FLO_TAX",
+    "FLO_MARK",
+    "PRC_RESID",
+    "NCAP_PKCNT",
+    "COM_PKRSV",
+    "COM_PKFLX",
+    "STG_EFF",
+    "STG_LOSS",
+    "STGIN_BND",
+    "PRC_ACTFLO",
+    "PRC_CAPACT",
+    "G_DRATE",
+    "G_YRFR",
+    "G_CUREX",
+    "IRE_FLO",
+    "IRE_FLOSUM",
+    "IRE_PRICE",
+    "SHAPE",
+    "UC_RHSRT",
+    "UC_RHSRTS",
+    "UC_RHSTS",
+    "UC_FLO",
+    "UC_ACT",
+    "UC_CAP",
+    "UC_NCAP",
+    "UC_COMPRD",
+    "UC_COMNET",
+    "UC_IRE",
+    "NCAP_DISC",
+    "VDA_FLOP",
+    "VDA_EMCB",
+    "VDA_CEH",
+    "FLO_EMIS",
+    "NCAP_AFC",
+    "NCAP_AFCS",
+    "ACT_EFF",
+]
```

Comment:
@Antti-L, this specifies the order in which TIMES sets and parameters would be printed in dd files. Do you have any comments?

Reply:
Well, why such an order?  For example, why is ACT_EFF the last one?  I think I would personally favour the category order 1) Domains, 2) Other Sets, 3) Parameters, and then each of these **alphabetically** within the same category. Domains must be first anyway (e.g. COM_GRP, COM, PRC, UC_N) to avoid GAMS errors, no?
 
And then, why so many missing parameters?  Are you not going to support all the TIMES parameters VEDA does (and are listed under Veda "Attributes")?

Reply:
Thanks @Antti-L! Originally @siddharth-krishna was parsing the times source code to ensure he could grab all the entities (and enable automated updating of the list for when TIMES source code changes) and preserve the order in which they are declared to avoid GAMS errors, but it got too complex. I don't remember what the current shortcut is based on.

Ok, so we can, just add a category property to the attribute master list and then populate it and keep it up to date manually. :-)

Reply:
Regarding the attributes list, it is incomplete because we are currently mostly including those attributes that are present in the test case. The master list will include all of the attributes, but we are still not sure how we want to structure it.

Reply:
Thanks both. Yes, I scraped this order by running Demo 1 via Veda and Gams and looking at the order of sets and parameters in the resulting GDX file. I thought the order was important because some definitions depended on others? If I were to order by category and alphabetically within categories, are such dependencies satisfied?

How do I determine which sets are domains and which are not?

And yes, this is currently just the parameters that we have added support for. We are working on expanding this to cover all the parameters in the Demo models.

Reply:
Yes, @siddharth-krishna, the dependencies should be satisfied with the categorisation approach. I'll probably just do it manually in the master file. If the current approach is working, let's proceed with it. We can then transition to using categories once we are using the master list.